File: superstitioMain/java/superstitio/cards/maso/SkillCard/cruelTorture/CruelTorture_Dismember.java
package superstitio.cards.maso.SkillCard.cruelTorture;

import basemod.helpers.CardModifierManager;
import com.megacrit.cardcrawl.actions.common.LoseHPAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import superstitio.DataManager;
import superstitio.cardModifier.modifiers.card.CruelTortureTag;
import superstitio.cards.maso.MasoCard;
import superstitio.delayHpLose.RemoveDelayHpLoseBlock;
import superstitioapi.hangUpCard.CardOrb;
import superstitioapi.hangUpCard.Card_TriggerHangCardManually;
import superstitioapi.hangUpCard.HangUpCardGroup;

//凌迟
public class CruelTorture_Dismember extends MasoCard implements Card_TriggerHangCardManually {
    public static final String ID = DataManager.MakeTextID(CruelTorture_Dismember.class);

    public static final CardType CARD_TYPE = CardType.SKILL;

    public static final CardRarity CARD_RARITY = CardRarity.COMMON;

    public static final CardTarget CARD_TARGET = CardTarget.SELF;

    private static final int COST = 1;
    private static final int MAGIC = 1;
    private static final int BLOCK = 3;
    private static final int UPGRADE_BLOCK = 1;


    public CruelTorture_Dismember() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        this.setupMagicNumber(MAGIC);
        this.setupBlock(BLOCK, UPGRADE_BLOCK, new RemoveDelayHpLoseBlock());
        CardModifierManager.addModifier(this, new CruelTortureTag());
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        HangUpCardGroup.forEachHangUpCard(orb -> {
            addToBot(new LoseHPAction(AbstractDungeon.player, AbstractDungeon.player, this.magicNumber));
            addToBot_gainBlock();
        }).addToBotAsAbstractAction();
    }

    @Override
    public void upgradeAuto() {
    }

    @Override
    public boolean forceFilterCardOrbToHoveredMode(CardOrb orb) {
        orb.targetType = CardOrb.HangOnTarget.Self;
        orb.actionType = CardOrb.HangEffectType.Good;
        return true;
    }

    @Override
    public int forceChangeOrbCounterShown(CardOrb orb) {
        return 0;
    }
}

File: superstitioMain/java/superstitio/cards/maso/SkillCard/cruelTorture/CruelTorture_Dismember.kt
package superstitio.cards.maso.SkillCard.cruelTorture

import basemod.helpers.CardModifierManager
import com.megacrit.cardcrawl.actions.common.LoseHPAction
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.monsters.AbstractMonster
import superstitio.DataManager
import superstitio.cardModifier.modifiers.card.CruelTortureTag
import superstitio.cards.IsMasoCard
import superstitio.cards.maso.MasoCard
import superstitio.delayHpLose.RemoveDelayHpLoseBlock
import superstitioapi.cards.addRetainMod
import superstitioapi.hangUpCard.CardOrb
import superstitioapi.hangUpCard.Card_TriggerHangCardManually
import superstitioapi.hangUpCard.HangUpCardGroup
@IsMasoCard
//凌迟
class CruelTorture_Dismember : MasoCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET), Card_TriggerHangCardManually
{
    init
    {
        this.setupMagicNumber(MAGIC)
        this.setupBlock(BLOCK, UPGRADE_BLOCK, RemoveDelayHpLoseBlock())
        CardModifierManager.addModifier(this, CruelTortureTag())
        this.addRetainMod()
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?)
    {
        HangUpCardGroup.forEachHangUpCard {
            addToBot(LoseHPAction(AbstractDungeon.player, AbstractDungeon.player, this.magicNumber))
            addToBot_gainBlock()
        }.addToBotAsAbstractAction()
    }

    override fun upgradeAuto()
    {
    }

    override fun forceFilterCardOrbToHoveredMode(orb: CardOrb): Boolean
    {
        orb.targetType = CardOrb.HangOnTarget.Self
        orb.actionType = CardOrb.HangEffectType.Good
        return true
    }

    override fun forceChangeOrbCounterShown(orb: CardOrb): Int
    {
        return 0
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(CruelTorture_Dismember::class.java)

        val CARD_TYPE: CardType = CardType.SKILL

        val CARD_RARITY: CardRarity = CardRarity.COMMON

        val CARD_TARGET: CardTarget = CardTarget.SELF

        private const val COST = 1
        private const val MAGIC = 1
        private const val BLOCK = 3
        private const val UPGRADE_BLOCK = 1
    }
}

